tests/ui/recursion_limit/invalid_digit.stderr

```diff
@@ -6,15 +6,5 @@ LL | #![recursion_limit = "-100"]
    |                      |
    |                      not a valid integer
 
-error: `limit` must be a non-negative integer
-  --> $DIR/invalid_digit.rs:3:1
-   |
```

Comment:
this doubled up error cannot be prevented. If we do, we'd get a similar situation as with crate_name where we either forget some cases, or delay as bug giving an ice when this literal happens to be a macro that before expansion is invalid but expanded *is* valid. e.g. `concat!("100")` which before expansion is an expression we should reject but after expansion is a perfectly valid integer literal